paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -113,154 +112,193 @@ class PaparazziPlugin : Plugin<Project> {
         it.description = "Run screenshot tests for variant '${variant.name}'"
       }
-      verifyVariants.configure { it.dependsOn(verifyTaskProvider) }
+    verifyVariants.configure { it.dependsOn(verifyTaskProvider) }
 
-      val isRecordRun = project.objects.property(Boolean::class.java)
-      val isVerifyRun = project.objects.property(Boolean::class.java)
+    val isRecordRun = project.objects.property(Boolean::class.java)
+    val isVerifyRun = project.objects.property(Boolean::class.java)
 
-      project.gradle.taskGraph.whenReady { graph ->
-        isRecordRun.set(recordTaskProvider.map { graph.hasTask(it) })
-        isVerifyRun.set(verifyTaskProvider.map { graph.hasTask(it) })
-      }
+    project.gradle.taskGraph.whenReady { graph ->
+      isRecordRun.set(recordTaskProvider.map { graph.hasTask(it) })
+      isVerifyRun.set(verifyTaskProvider.map { graph.hasTask(it) })
+    }
 
-      val testTaskProvider = project.tasks.named("test$testVariantSlug", Test::class.java) { test ->
-        test.systemProperties["paparazzi.test.resources"] =
-          writeResourcesTask.flatMap { it.paparazziResources.asFile }.get().path
-        test.systemProperties["paparazzi.project.dir"] = projectDirectory.toString()
-        test.systemProperties["paparazzi.build.dir"] = buildDirectory.get().toString()
-        test.systemProperties["paparazzi.report.dir"] = reportOutputDir.get().toString()
-        test.systemProperties["paparazzi.snapshot.dir"] = snapshotOutputDir.toString()
-        test.systemProperties["paparazzi.artifacts.cache.dir"] = gradleUserHomeDir.path
-        test.systemProperties["kotlinx.coroutines.main.delay"] = true
-        test.systemProperties.putAll(project.properties.filterKeys { it.startsWith("app.cash.paparazzi") })
-
-        test.inputs.property("paparazzi.test.record", isRecordRun)
-        test.inputs.property("paparazzi.test.verify", isVerifyRun)
-
-        test.inputs.dir(mergeResourcesOutputDir)
-        test.inputs.dir(mergeAssetsOutputDir)
-        test.inputs.files(nativePlatformFileCollection)
-          .withPropertyName("paparazzi.nativePlatform")
-          .withPathSensitivity(PathSensitivity.NONE)
-
-        test.outputs.dir(reportOutputDir)
-        test.outputs.dir(snapshotOutputDir)
-
-        test.doFirst {
-          // Note: these are lazy properties that are not resolvable in the Gradle configuration phase.
-          // They need special handling, so they're added as inputs.property above, and systemProperty here.
-          test.systemProperties["paparazzi.platform.data.root"] =
-            nativePlatformFileCollection.singleFile.absolutePath
-          test.systemProperties["paparazzi.test.record"] = isRecordRun.get()
-          test.systemProperties["paparazzi.test.verify"] = isVerifyRun.get()
-        }
+    val testTaskName = "test$testVariantSlug"
+    project.tasks.matching {
+      it is Test && it.name == testTaskName
+    }.configureEach { rawTest ->
```

Comment:
Similar consistency as above:
```suggestion
    project.tasks
      .matching { it is Test && it.name == testTaskName }
      .configureEach { rawTest ->
```

Reply:
Same answer as above